### DISCUSSION TITLE
How to save user to public schema and update role while signing up?

### DISCUSSION THREAD

**Discussion by roker15** [supabase/supabase#3243] [Questions] [answered]
Here is the previsous [discussion](https://github.com/supabase/supabase/discussions/458)  , answer has been marked accepted but whosoever is asked is not satisfied and answer is still pending since 8 -jan. so i have to ask this again
``
How to save user to public schema and update role while signing up? folloing is not working and user is undefined.
```
const { user, error } = await supabase.auth.signIn({ email });
      console.log(console.log(user)) //user is undefined here
const { data } = await supabase.from<Profile>("profiles").upsert({id:user?.id,role:role });
```

**Comment by MildTomato** [comment, marked as answer]
Can we keep the discussion in the other thread? So it’s easier for others to find answers to queries.

If threads have new replies they are automatically bumped up the discussion board and will get picked up again - you don’t need to open up a new thread.
